src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -133,27 +163,96 @@ internal static async Task<int> Run(PackageUpdateArgs args, ILoggerWithColor log
     {
         if (packages is null || packages.Count == 0)
         {
-            logger.LogMinimal(Strings.Unsupported_UpgradeAllPackages, ConsoleColor.Red);
-            return (null, null);
+            return await GetAllPackagesWithUpdatesAsync(project, versionChooser, settings, logger, cancellationToken);
+        }
+
+        var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
+        if (sourceMapping.IsEnabled)
+        {
+            logger.LogMinimal(Strings.Unsupported_PackageSourceMapping, ConsoleColor.Red);
+            return new List<PackageUpdateResult>();
         }
 
-        if (packages.Count > 1)
+        var packagesToUpdate = new List<PackageUpdateResult>();
+        bool hasErrors = false;
+
+        foreach (var package in packages)
         {
-            logger.LogMinimal(Strings.Unsupported_MoreThanOnePackage, ConsoleColor.Red);
-            return (null, null);
+            var result = await GetSinglePackageToUpdateAsync(package, project, versionChooser, settings, logger, suppressWarnings: false, cancellationToken);
+            if (result != null)
+            {
+                packagesToUpdate.Add(result);
+            }
+            else
+            {
+                hasErrors = true;
+            }
         }
 
+        return hasErrors ? new List<PackageUpdateResult>() : packagesToUpdate;
+    }
+
+    private static async Task<List<PackageUpdateResult>> GetAllPackagesWithUpdatesAsync(
+        PackageSpec project,
+        IVersionChooser versionChooser,
+        ISettings settings,
+        ILoggerWithColor logger,
+        CancellationToken cancellationToken)
+    {
         var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
         if (sourceMapping.IsEnabled)
         {
             logger.LogMinimal(Strings.Unsupported_PackageSourceMapping, ConsoleColor.Red);
-            return (null, null);
+            return new List<PackageUpdateResult>();
         }
 
-        var package = packages.Single();
+        var allProjectPackages = GetAllPackagedReferencedByProject(project);
+        var packagesToUpdate = new List<PackageUpdateResult>();
+
+        foreach (var packageId in allProjectPackages)
+        {
+            var packageToCheck = new Package { Id = packageId, VersionRange = null };
+            var result = await GetSinglePackageToUpdateAsync(packageToCheck, project, versionChooser, settings, logger, suppressWarnings: true, cancellationToken);
+            if (result != null)
```

Comment:
In the above private method `GetPackagesToUpdateAsync`, when `result == null` we set `hasErrors = true` and return an empty list. Is it intentional to not do the same thing here?